test/powershell/engine/Module/ModulePath.Tests.ps1

```diff
@@ -188,3 +188,53 @@ Describe "SxS Module Path Basic Tests" -tags "CI" {
         }
     }
 }
+
+# need to be elevated because we are installing a module in the allusers scope
+Describe "ModuleIntrinsics.GetPSModulePath API tests" -Tags @("CI","RequireAdminOnWindows","RequireSudoOnUnix") {
+    BeforeAll {
+        # create a local repostory and install a module
+        $localSourceName = [Guid]::NewGuid().ToString("n")
+        $localSourceLocation = Join-Path $PSScriptRoot assets
+        Register-PSRepository -Name $localSourceName -SourceLocation $localSourceLocation -InstallationPolicy Trusted -ErrorAction SilentlyContinue
+        Install-Module -Force -Scope AllUsers -Name PowerShell.TestPackage -Repository $localSourceName -ErrorAction SilentlyContinue
+        Install-Module -Force -Scope CurrentUser -Name PowerShell.TestPackage -Repository $localSourceName -ErrorAction SilentlyContinue
+
+        $testCases = @(
+            @{ Name = "User"   ; Expected = $IsWindows ?
+                (Resolve-Path ([Environment]::GetFolderPath("Personal") + "\PowerShell\Modules")).Path :
+                (Resolve-Path ([System.Management.Automation.Platform]::SelectProductNameForDirectory("USER_MODULES"))).Path
+                }
+            @{ Name = "Shared" ; Expected = $IsWindows ?
+                [Environment]::GetFolderPath("ProgramFiles") + "\PowerShell\Modules" :
+                (Resolve-Path ([System.Management.Automation.Platform]::SelectProductNameForDirectory("SHARED_MODULES"))).Path
+            }
+            @{ Name = "PSHome" ; Expected = (Resolve-Path (Join-Path $PSHOME Modules)).Path }
+        )
+        # resolve the paths to ensure they are in the correct format
+        $currentModulePathElements = $env:PSModulePath -split [System.IO.Path]::PathSeparator | Foreach-Object { (Resolve-Path $_).Path }
+    }
+
+    AfterAll {
+        Unregister-PSRepository -Name $localSourceName -ErrorAction SilentlyContinue
+    }
+
+    It "The GetPSModulePath api is available" {
+        { [System.Management.Automation.ModuleIntrinsics]::GetPSModulePath("PSHOME") } | Should -not -throw
```

Comment:
I don't think this specific test case is needed since it's implicitly covered below.

Reply:
removed